modulemd/v2/include/modulemd-2.0/private/test-utils.h

```diff
@@ -15,9 +15,30 @@
 
 #include <glib.h>
 #include <locale.h>
+#include <yaml.h>
 
 G_BEGIN_DECLS
 
+typedef struct _CommonMmdTestFixture
+{
+} CommonMmdTestFixture;
+
+extern int modulemd_test_signal;
+
+
+/**
+ * modulemd_test_signal_handler:
+ * @sig_num: The signal received
+ *
+ * Sets the global variable modulemd_test_signal with the value of the signal
+ * that was received.
+ *
+ * Since: 2.0
+ */
+void
+modulemd_test_signal_handler (int sig_num);
```

Comment:
Given how often we try to run some code to check that it emits a signal, maybe it's an idea to instead make a macro that takes the signal and a function to run and sets up and asserts the signal handler?
Kind of like the `expect_signal()` in the python `base.py`?

Reply:
Yeah, I was thinking the same thing. I'll look into that in a follow-up patch.